tests/pytests/test_empty.py

```diff
@@ -3,6 +3,62 @@
 
 EMPTY_RESULT = [0]
 
+def TestEmptyNonIndexed():
+    """Tests that we throw and error in case of a query with an empty string
+    for a field that doesn't index empty values."""
+
+    env = DialectEnv()
+    conn = getConnectionByEnv(env)
+
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'tag', 'TAG', 'text', 'TEXT').ok()
+    conn.execute_command('HSET', 'h2', 'tag', '', 'text', '')
+
+    MAX_DIALECT = set_max_dialect(env)
+    for dialect in range(2, MAX_DIALECT + 1):
+        env.set_dialect(dialect)
+
+        # A query with no field mask should return an empty result, not throwing
+        # an error
+        res = conn.execute_command('FT.SEARCH', 'idx', '')
+        env.assertEqual(res, EMPTY_RESULT)
+        res = conn.execute_command('FT.SEARCH', 'idx', '""')
+        env.assertEqual(res, EMPTY_RESULT)
+        res = conn.execute_command('FT.SEARCH', 'idx', "''")
+        env.assertEqual(res, EMPTY_RESULT)
+
+        # Any query containing "@tag:{''}" or "@text:''" should throw an error
+        for query in ['@tag:{""}', "@tag:{''}", '@text:""', "@text:''"]:
```

Comment:
Bug? Consider testing with two tag fields:
```sh
127.0.0.1:6379> FT.CREATE idx schema t1 tag indexempty t2 tag
OK
127.0.0.1:6379> FT.SEARCH idx '@t1:("")' dialect 2
(error) Empty value for field that does not index empty values
127.0.0.1:6379> FT.SEARCH idx '@t2:("")' dialect 2
(error) Empty value for field that does not index empty values
```

Reply:
Right!
Corrected this back to return `[0]`, as before, in order not to break (this should be an error message..).